src/ConfigCatClient/Evaluation/EvaluateLogHelper.cs

```diff
@@ -0,0 +1,346 @@
+using ConfigCat.Client.Utils;
+using System;
+using System.Globalization;
+
+namespace ConfigCat.Client.Evaluation;
+
+internal static class EvaluateLogHelper
+{
+    public const string InvalidItemPlaceholder = "<invalid item>";
+    public const string InvalidNamePlaceholder = "<invalid name>";
+    public const string InvalidOperatorPlaceholder = "<invalid operator>";
+    public const string InvalidReferencePlaceholder = "<invalid reference>";
+    public const string InvalidValuePlaceholder = "<invalid value>";
+
+    private const int StringListMaxLength = 10;
+
+    public static IndentedTextBuilder AppendEvaluationResult(this IndentedTextBuilder builder, bool result)
+    {
+        return builder.Append(result ? "true" : "false");
+    }
+
+    private static IndentedTextBuilder AppendComparisonCondition(this IndentedTextBuilder builder, string? comparisonAttribute, Comparator comparator, object? comparisonValue)
+    {
+        return builder.Append($"User.{comparisonAttribute} {comparator.ToDisplayText()} '{comparisonValue ?? InvalidValuePlaceholder}'");
+    }
+
+    private static IndentedTextBuilder AppendComparisonCondition(this IndentedTextBuilder builder, string? comparisonAttribute, Comparator comparator, string? comparisonValue, bool isSensitive = false)
+    {
+        return builder.AppendComparisonCondition(comparisonAttribute, comparator, !isSensitive ? (object?)comparisonValue : "<hashed value>");
+    }
+
+    private static IndentedTextBuilder AppendComparisonCondition(this IndentedTextBuilder builder, string? comparisonAttribute, Comparator comparator, string[]? comparisonValue, bool isSensitive = false)
+    {
+        if (comparisonValue is null)
+        {
+            return builder.AppendComparisonCondition(comparisonAttribute, comparator, (object?)null);
+        }
+
+        const string valueText = "value", valuesText = "values";
+
+        if (isSensitive)
+        {
+            return builder.Append($"User.{comparisonAttribute} {comparator.ToDisplayText()} [<{comparisonValue.Length} hashed {(comparisonValue.Length == 1 ? valueText : valuesText)}>]");
+        }
+        else
+        {
+            var comparisonValueFormatter = new StringListFormatter(comparisonValue, StringListMaxLength, getOmittedItemsText: static count =>
+                string.Format(CultureInfo.InvariantCulture, " ... <{0} more {1}>", count, count == 1 ? valueText : valuesText));
```

Comment:
If I remember right for list we would like to use `()`
```suggestion
                string.Format(CultureInfo.InvariantCulture, " ... ({0} more {1})", count, count == 1 ? valueText : valuesText));
```